api/envoy/service/discovery/v3/discovery.proto

```diff
@@ -276,4 +290,17 @@ message Resource {
   // Cache control properties for the resource.
   // [#not-implemented-hide:]
   CacheControl cache_control = 7;
+
+  // The subscription ID(s) that this resource is associated with, if
+  // sent in the request when the client subscribes to a new resource.
+  // This needs to be populated only when (a) a resource is first sent
+  // to the client or (b) when the client subscribes to a new resource
+  // locator that matches a resource that the client is already
+  // subscribing to.
+  // Note that in case (b), the resource field does not need to be
+  // populated. This allows the server to tell the client that a
+  // resource that it had already been sent (because it matched a
+  // resource locator that the client had previously subscribed to) also
+  // matches another resource locator that the client has newly subscribed to.
+  repeated uint64 new_subscription_ids = 8;
```

Comment:
How common will it be to have two different resource requests that match the same returned resource?

Reply:
I think it will be fairly common.  See the section about cache behavior in the design doc.  If the cache has one client subscription for `...?foo=@range[0,1]` and another client subscription for `...?foo=@range[1,2]`, a resource `...?foo=1` matches both of those subscriptions.